### PR TITLE
Ticket 3736

### PR DESCRIPTION
**Pull request by boeuf-codes**
### Description of work

- Added action to the block context menu to navigate to run control settings
- Edited title in block configuration dialogue to clarify runcontrol settings there are _default_  settings rather than current.

### Ticket

[*Link to Ticket #3736*](https://github.com/ISISComputingGroup/IBEX/issues/3736)

### Acceptance criteria

- be able to open the runcontrol settings menu via the context menu of a block.
- make clearer the run control settings entered there are the default runcontrol values, to help make the distinction between those and the current runcontrol settings accessed via the menu bar.

### Unit/System tests
n/a

### Manual tests
See test: **section 20, test 27**

---

#### Code Review

- [ ] Is the code of an acceptable quality?
- [ ] If the change is to an OPI, does the `check_opi_format.py` script in [C:\Instrument\Dev\ibex_gui\base\uk.ac.stfc.isis.ibex.opis](https://github.com/ISISComputingGroup/ibex_gui/blob/master/base/uk.ac.stfc.isis.ibex.opis/check_opi_format.py) pass?
- [ ] Do the changes function as described and is it robust?
- [ ] Is there associated PR for the [release notes](https://github.com/ISISComputingGroup/IBEX/blob/master/release_notes/ReleaseNotes_Upcoming.md)?

### Final Steps
- [ ] Reviewer has also merged the [release notes](https://github.com/ISISComputingGroup/IBEX/blob/master/release_notes/ReleaseNotes_Upcoming.md)

### CODE REVIEW

File: base/uk.ac.stfc.isis.ibex.ui.blocks/src/uk/ac/stfc/isis/ibex/ui/blocks/groups/BlocksMenu.java [hunk 1]
```diff
@@ -113,12 +120,27 @@ public void run() {
      *
      * @param displayBlock the selected block
      */
-    public BlocksMenu(DisplayBlock displayBlock) {
+    public BlocksMenu(DisplayBlock displayBlock, IHandlerService handlerService) {
```
Comment (JackEAllen) [inline base/uk.ac.stfc.isis.ibex.ui.blocks/src/uk/ac/stfc/isis/ibex/ui/blocks/groups/BlocksMenu.java:123]: Can we add `handlerService` to the docstring?

File: base/uk.ac.stfc.isis.ibex.ui.blocks/src/uk/ac/stfc/isis/ibex/ui/blocks/groups/BlocksMenu.java [hunk 2]
```diff
@@ -185,6 +207,8 @@ public void run() {
                 new EditBlockHandler(block.getName()).execute(null); //TODO e4 migrate: This will be added as a command which includes a shell at that time make this correct
             }
         };
+        
+
```
Comment (JackEAllen) [inline base/uk.ac.stfc.isis.ibex.ui.blocks/src/uk/ac/stfc/isis/ibex/ui/blocks/groups/BlocksMenu.java:211]: NITPICK: Could we remove this dead space at the end?

File: base/uk.ac.stfc.isis.ibex.ui.blocks/src/uk/ac/stfc/isis/ibex/ui/blocks/groups/BlocksMenu.java [hunk 3]
```diff
@@ -113,12 +120,27 @@ public void run() {
...
      */
-    public BlocksMenu(DisplayBlock displayBlock) {
+    public BlocksMenu(DisplayBlock displayBlock, IHandlerService handlerService) {
 		this.block = displayBlock;
 		
 		Configurations.getInstance().server().setCurrentConfig().addOnCanWriteChangeListener(readOnlyListener);
 
         add(new GroupMarker(BLOCK_MENU_GROUP));
+          
+        final IAction viewRunControlSettingsAction = new Action(VIEW_RUN_CONTROL_SETTINGS) {
```
Comment (JackEAllen) [inline base/uk.ac.stfc.isis.ibex.ui.blocks/src/uk/ac/stfc/isis/ibex/ui/blocks/groups/BlocksMenu.java:130]: Do you think it's worth adding a brief comment here just to describe the below block?

File: base/uk.ac.stfc.isis.ibex.ui.blocks/src/uk/ac/stfc/isis/ibex/ui/blocks/groups/Group.java [hunk 4]
```diff
@@ -84,7 +85,7 @@ public class Group extends Composite {
 	 * @param panel
 	 *            The panel that shows the groups and blocks.
 	 */
-	public Group(Composite parent, int style, DisplayGroup group, GroupsPanel panel) {
+	public Group(Composite parent, int style, DisplayGroup group, GroupsPanel panel, IHandlerService handlerService) {
```
Comment (JackEAllen) [inline base/uk.ac.stfc.isis.ibex.ui.blocks/src/uk/ac/stfc/isis/ibex/ui/blocks/groups/Group.java:88]: Can we update the docstring to include `handlerService`

File: base/uk.ac.stfc.isis.ibex.ui.blocks/src/uk/ac/stfc/isis/ibex/ui/blocks/groups/GroupsPanel.java [hunk 5]
```diff
@@ -87,8 +89,9 @@ private enum ConnectionStatus {
      * @param style
      *            The SWT style of the panel.
      */
-	public GroupsPanel(Composite parent, int style) {
+	public GroupsPanel(Composite parent, int style, IHandlerService handlerService) {
```
Comment (JackEAllen) [inline base/uk.ac.stfc.isis.ibex.ui.blocks/src/uk/ac/stfc/isis/ibex/ui/blocks/groups/GroupsPanel.java:92]: Can we update the docstring to include `handlerService`

File: base/uk.ac.stfc.isis.ibex.ui.blocks/src/uk/ac/stfc/isis/ibex/ui/blocks/views/BlocksView.java [hunk 6]
```diff
@@ -110,7 +111,7 @@ private void setUnknownConfiguration() {
 	 * @param modelService The E4 service responsible for handling model elements
 	 */
     @PostConstruct
-	public void createPartControl(final Composite parent, MApplication app, EModelService modelService) {
+	public void createPartControl(final Composite parent, MApplication app, EModelService modelService, IHandlerService handlerService) {
```
Comment (JackEAllen) [inline base/uk.ac.stfc.isis.ibex.ui.blocks/src/uk/ac/stfc/isis/ibex/ui/blocks/views/BlocksView.java:114]: Can we update the docstring to include `handlerService`

**Review by JackEAllen — CHANGES_REQUESTED** [review]
Thanks for working on this. From testing on the GUI, your changes act as expected 🥇
 
Overall this is really good. Mainly just needs a merge conflict resolving. I have left a few minor changes and also the odd nit-picks which you're more than welcome to decide on whether you agree with or not. 
I'm going to place this into rework so you can make the minor changes I've suggested and fix the merge conflict, then I'm happy to pick it back up to review again after.

File: base/uk.ac.stfc.isis.ibex.ui.runcontrol/src/uk/ac/stfc/isis/ibex/ui/runcontrol/dialogs/EditRunControlDialog.java [hunk 7]
```diff
@@ -43,26 +44,24 @@
...
 	private final RunControlViewModel viewModel;
+	private static final String TITLE = "Run-Control Settings";
 	
 	/**
 	 * Creates a dialog for configuring the run-control settings.
 	 * 
 	 * @param parentShell the parent SWT Shell
-	 * @param title the title of the dialog window
 	 * @param configServer the Config Server 
 	 * @param runControlServer the Run Control server
```
[the remark this answers is not in the document]
Comment (boeuf-codes) [reply base/uk.ac.stfc.isis.ibex.ui.runcontrol/src/uk/ac/stfc/isis/ibex/ui/runcontrol/dialogs/EditRunControlDialog.java]: It seems I had already moved these to the correct place but not deleted here, so have just removed.

File: base/uk.ac.stfc.isis.ibex.ui.blocks/src/uk/ac/stfc/isis/ibex/ui/blocks/groups/BlocksMenu.java (hunk 1 above)
Comment (JackEAllen) [reply base/uk.ac.stfc.isis.ibex.ui.blocks/src/uk/ac/stfc/isis/ibex/ui/blocks/groups/BlocksMenu.java:124]: Thanks for adding this. I would say it describes `handlerService` pretty well.

**Comment by boeuf-codes** [comment]
Commits for rework:
- Docstrings & comment adding https://github.com/ISISComputingGroup/ibex_gui/pull/1452/commits/2b24de31d3c11f6bcbf46c8348f2604a0cb0a4a0
-  Merging master https://github.com/ISISComputingGroup/ibex_gui/pull/1452/commits/de49d8771bcd423d03b649f379066818e0ddeb80

**Review by JackEAllen — APPROVED** [review]
Thanks for making these changes and merging in the conflict. Have tested again and happy to put this through now 👍

**Pull request merged**